src/main/java/org/example/SimpleSink.java

```diff
@@ -0,0 +1,56 @@
+package org.example;
+
+import io.numaproj.numaflow.sinker.Datum;
+import io.numaproj.numaflow.sinker.DatumIterator;
+import io.numaproj.numaflow.sinker.Response;
+import io.numaproj.numaflow.sinker.ResponseList;
+import io.numaproj.numaflow.sinker.Server;
+import io.numaproj.numaflow.sinker.Sinker;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * This is a simple User Defined Sink example which logs the input message
+ */
+
+@Slf4j
+public class SimpleSink extends Sinker {
+
+    public static void main(String[] args) throws Exception {
+        Server server = new Server(new SimpleSink());
+
+        // Start the server
+        server.start();
+
+        // wait for the server to shut down
+        server.awaitTermination();
+    }
+
+    @Override
+    public ResponseList processMessages(DatumIterator datumIterator) {
+        ResponseList.ResponseListBuilder responseListBuilder = ResponseList.newBuilder();
+        while (true) {
+            Datum datum = null;
+            try {
+                datum = datumIterator.next();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                continue;
+            }
+            // null means the iterator is closed, so we break the loop
+            if (datum == null) {
```

Comment:
This pattern indicates that the `while` loop could be replaced with a `do .. while (datum != null)` loop.

Reply:
Just realized that you needed to check if datum is null before adding the response. In this case, let's keep your while loop.